File: Java/Bikes/src/Bicycle.java
public class Bicycle {

    // this is the base class
    // the Bicycle class has two fields
    private int gear; // change from website
    private int speed; // chnange from website

    public Bicycle(int gear, int speed) {
        this.gear = gear;
        this.speed = speed;
    }

    public void applyBrake(int decrement) {
        speed -= decrement;
    }

    public void speedUp(int increment) {
        speed += increment;
    }

    @Override
    public String toString() {
        return ("No of gears are " + gear + "\n"
                + "speed of bicycle is " + speed);
    }

    @Override
    public boolean equals(Object obj) {
        Bicycle b2 = (Bicycle) obj;
        return this.gear == b2.gear && this.speed == b2.speed;
    }

}

File: Python/Bikes/Bicycle.py
class Bicycle:

    __gear =  None 
    __speed = None

    def __init__(self,gear,speed):
        self.__gear = gear
        self.__speed = speed
    
    def applyBrake(self, decrement):
        self.__speed -= decrement
        
    def speedUp(self, increment): 
        self.__speed += increment
        

    def __str__(self):
        return "No of gears are " + str(self.__gear) + "\n" \
                + "speed of bicycle is " + str(self.__speed)
                
    def  __eq__(self,b2):
        return self.__gear == b2.__gear and self.__speed == b2.__speed
